### ISSUE TITLE
detect-aws-credentials false positives for empty AWS variables

### ISSUE THREAD

**Issue by begoon** [pre-commit/pre-commit-hooks#449]
When there are AWS credentials variables with empty values are in the environment, for example:
```
AWS_SECRET_ACCESS_KEY=
AWS_ACCESS_KEY_ID=
```

`detect_aws_credentials` defects a false positive issue, for example:

```
Detect AWS Credentials...................................................Failed
hookid: detect-aws-credentials

AWS secret found in some/file/admin.py: ****************************
```

It happens because the `key` value is an empty string "", and obviously the empty values "" can be "detected" in every file.

The following fix allows to skip searching for empty values of the keys.
```
index fe18f4d..aae0734 100644
--- a/pre_commit_hooks/detect_aws_credentials.py
+++ b/pre_commit_hooks/detect_aws_credentials.py
@@ -84,7 +84,7 @@ def check_file_for_aws_keys(
             for key in keys:
                 # naively match the entire file, low chance of incorrect
                 # collision
-                if key in text_body:
+                if key and key in text_body:
                     bad_files.append(BadFile(filename, key[:4].ljust(28, '*')))
     return bad_files
```

**Comment by begoon** [comment]
Version 2.5.0.

**Comment by WhyNotHugo** [comment]
AFAIK, all valid aws keys start with `AKIA`, so that might also be a more reliable hint to detect them.

**Comment by asottile** [comment]
this was already resolved in #450
